protocol/x/clob/types/clob_pair.go

```diff
@@ -10,9 +10,15 @@ import (
 // may be transitioned to from this state. Note the keys of this map may be
 // a subset of the types defined in the proto for ClobPair_Status.
 var SupportedClobPairStatusTransitions = map[ClobPair_Status]map[ClobPair_Status]struct{}{
-	ClobPair_STATUS_ACTIVE: {},
+	ClobPair_STATUS_ACTIVE: {
+		ClobPair_STATUS_FINAL_SETTLEMENT: struct{}{},
+	},
 	ClobPair_STATUS_INITIALIZING: {
-		ClobPair_STATUS_ACTIVE: struct{}{},
+		ClobPair_STATUS_ACTIVE:           struct{}{},
+		ClobPair_STATUS_FINAL_SETTLEMENT: struct{}{},
+	},
+	ClobPair_STATUS_FINAL_SETTLEMENT: {
+		ClobPair_STATUS_INITIALIZING: struct{}{},
```

Comment:
Why is this a valid state transition?

For simplicity, is it fine to assume that the final settlement state is terminal for markets? Seems safer to just re-initialize a new CLOB pair.

Note if we allow this, then it's possible for a final settlement market to eventually begin trading again which is difficult to test (requires winding down market then bringing it up again which would require multiple gov proposals right?)

Reply:
I took the stance that the flow for re-enabling trading for a market would be to transition back into initializing mode, and then to active mode. 

The reasoning here is that this project does not cover cleaning up the clob pair from state (removing from state). This is because we don't want to allow the clob pair to be deleted and then a new clob pair to be created with the same id as the deleted one. This would have implications, particularly for indexer. 

Because we are going to leave the clob pair in state, it makes sense to allow trading to be re-enabled with a MsgUpdateClobPair message instead of creating a new clob pair

Reply:
>The reasoning here is that this project does not cover cleaning up the clob pair from state (removing from state). This is because we don't want to allow the clob pair to be deleted and then a new clob pair to be created with the same id as the deleted one. This would have implications, particularly for indexer.

Why not just create a new CLOB pair for trading the market, and leave the other one permanently in state?

I personally don't think it makes sense to support this functionality since it overcomplicates things. I think we should keep it simple as possible to minimize the surface area of this change.

Note that when markets are de-listed, they are essentially only wound back-up with significant changes (for example LUNA => LUNA classic). It is probably fine to create a new CLOB pair for this usecase.

LMK wyt.

Reply:
I see what you're saying, and this was my initial thought as well but during spec review process I think I was nudged the other direction.

To support your suggestion we would need to remove the current clob pair id from the clob keeper's `PerpetualIdToClobPairId` in-memory mapping of perpetual id -> clob pair. This is because we currently only support one clob pair per perpetual. If we want to be able to create a new one then the old one needs to be removed.

We would also need to update the hydration logic on startup to not populate Clob Pairs with final settlement status into this in-memory mapping.

I think this is kind of strange because the in-memory data structure (1 clob pair) would be a modified view of what is actually in state (2 clob pairs)

Reply:
also in slack you listed the concern of what happens if all positions haven't been closed by the time the market re-opens, but this problem still stands if we create a new clob pair because the clob pairs would in theory be pointing to the same perpetual so any trading in the new clob pair would be altering position size of the final settlement positions from the other clob pair